### PR TITLE
Make flex reorder, flow reorder and reparenting work with fragments

### PR DESCRIPTION
**Pull request by bkrmendy**
# Try it [here](https://utopia.pizza/p/e07b6893-cheerful-alarm/?branch_name=fix-reorder-when-fragmen-is-sibling)

## Description
This PR fills out `parentLayoutSystem`, `parentFlexDirection`, and `immediateParentBounds` on elements that don't have it defined, because they don't exist in the rendered DOM, and the DOM walker cannot measure them. 

This change makes fragments work with the following existing features:
- reorder (both flex and flow)
- keyboard reorder
- reparenting a fragment into another element

### CODE REVIEW

File: editor/src/components/canvas/canvas-strategies/strategies/flow-reorder-strategy.spec.browser2.tsx [hunk 1]
```diff
@@ -217,6 +286,8 @@ describe('Flow Reorder Strategy (Mixed Display Type)', () => {
     ])
 
     await renderResult.getDispatchFollowUpActionsFinished()
+
+    await wait(10000)
```
Comment (seanparsons) [inline editor/src/components/canvas/canvas-strategies/strategies/flow-reorder-strategy.spec.browser2.tsx:290]: Is this necessary? As it usually indicates we're not `await`ing something somewhere.

File: editor/src/components/canvas/canvas-strategies/strategies/flow-reorder-strategy.spec.browser2.tsx [hunk 2]
```diff
@@ -428,4 +499,31 @@ describe('Flow Reorder Strategy (Mixed Display Type)', () => {
...
+        'utopia-storyboard-uid/scene-aaa/app-entity',
+        'utopia-storyboard-uid/scene-aaa/app-entity:container',
+        'utopia-storyboard-uid/scene-aaa/app-entity:container/aaa',
+        'utopia-storyboard-uid/scene-aaa/app-entity:container/38e/bbb',
+        'utopia-storyboard-uid/scene-aaa/app-entity:container/ccc',
+      ])
+
+      await renderResult.getDispatchFollowUpActionsFinished()
+
+      await wait(10000)
```
Comment (seanparsons) [inline editor/src/components/canvas/canvas-strategies/strategies/flow-reorder-strategy.spec.browser2.tsx:523]: 10 seconds!

File: editor/src/components/canvas/canvas-strategies/strategies/flow-reorder-strategy.spec.browser2.tsx (hunk 1 above)
Comment (bkrmendy) [reply editor/src/components/canvas/canvas-strategies/strategies/flow-reorder-strategy.spec.browser2.tsx]: I forgot to delete this after debugging tests, I removed them in https://github.com/concrete-utopia/utopia/pull/3302/commits/a01cb2d89c9dd09f7b47016a21b4c70d2256b64b

File: editor/src/components/canvas/canvas-strategies/strategies/flow-reorder-strategy.spec.browser2.tsx (hunk 2 above)
Comment (bkrmendy) [reply editor/src/components/canvas/canvas-strategies/strategies/flow-reorder-strategy.spec.browser2.tsx]: I forgot to delete this after debugging tests, I removed them in https://github.com/concrete-utopia/utopia/pull/3302/commits/a01cb2d89c9dd09f7b47016a21b4c70d2256b64b

File: editor/src/components/canvas/canvas-strategies/strategies/flex-reorder-strategy.spec.browser2.tsx [hunk 3]
```diff
@@ -434,7 +663,6 @@ describe('Flex Reorder Strategy', () => {
 
       const startPoint = windowPoint({ x: targetElementBounds.x + 5, y: targetElementBounds.y + 5 })
       await mouseClickAtPoint(canvasControlsLayer, startPoint, { modifiers: cmdModifier })
-      await pressKey('Escape')
```
Comment (balazsbajorics) [inline editor/src/components/canvas/canvas-strategies/strategies/flex-reorder-strategy.spec.browser2.tsx:437]: why this change?

Comment (bkrmendy) [reply editor/src/components/canvas/canvas-strategies/strategies/flex-reorder-strategy.spec.browser2.tsx:437]: as far as I could see, if I press `esc` after selecting an element, the parent is selected, and I wanted to select the actual element that the test clicks on

File: editor/src/core/model/element-metadata-utils.ts [hunk 4]
```diff
@@ -1839,6 +1838,57 @@ function fillSpyOnlyMetadataWithFramesFromChildren(
...
+    const immediateParentBoundsFromChildren = childrenFromWorking.map(
+      (c) => c.specialSizeMeasurements.immediateParentBounds,
+    )
+
+    workingElements[pathStr] = {
+      ...spyElem,
+      ...sameThingFromWorkingElems,
+      specialSizeMeasurements: {
+        ...spyElem.specialSizeMeasurements,
+        parentLayoutSystem: allElemsEqual(parentLayoutSystemFromChildren)
```
Comment (balazsbajorics) [inline editor/src/core/model/element-metadata-utils.ts:1879]: would it be nice to have a helper function for "if all elements equal, use their value, otherwise fall back to..."? or is this readable and good enough?

File: editor/src/components/canvas/canvas-strategies/strategies/keyboard-reorder-strategy.spec.browser2.tsx [hunk 5]
```diff
@@ -100,6 +107,61 @@ const TestProjectMixedInlineFlow = `
...
+      <div
+        style={{
+          backgroundColor: '#00abff',
+          width: 123,
+          height: 215,
+          contain: 'layout',
+        }}
+        data-uid='child1'
+      />
+      <React.Fragment>
```
Comment (balazsbajorics) [inline editor/src/components/canvas/canvas-strategies/strategies/keyboard-reorder-strategy.spec.browser2.tsx:133]: you can give this an explicit data-uid! – it'll throw a react warning but so be it.

Comment (bkrmendy) [reply editor/src/components/canvas/canvas-strategies/strategies/keyboard-reorder-strategy.spec.browser2.tsx]: Thanks! I updated the test in https://github.com/concrete-utopia/utopia/pull/3302/commits/6d6101dcc1cf0132fea275c33a86187185426f07

**Review by balazsbajorics — APPROVED** [review]

File: editor/src/core/model/element-metadata-utils.ts (hunk 4 above)
Comment (bkrmendy) [reply editor/src/core/model/element-metadata-utils.ts:1879]: (resolving, we agreed to leave it as is, since the code is still simple enough to read)

**Review by seanparsons — APPROVED** [review]
